pkg/promotion/runner/builtin/argocd_updater.go

```diff
@@ -690,38 +788,99 @@ func (a *argocdUpdater) logAppEvent(
 	}
 }
 
-// getAuthorizedApplication returns an Argo CD Application in the given namespace
-// with the given name, if it is authorized for mutation by the Kargo Stage
-// represented by stageMeta.
-func (a *argocdUpdater) getAuthorizedApplication(
+// getAuthorizedApplications returns a slice of Argo CD Applications that match
+// the given update specification (either by name or by label selector) and are
+// authorized for mutation by the Kargo Stage.
+func (a *argocdUpdater) getAuthorizedApplications(
 	ctx context.Context,
 	stepCtx *promotion.StepContext,
-	appKey client.ObjectKey,
-) (*argocd.Application, error) {
-	app, err := argocd.GetApplication(
-		ctx,
-		a.argocdClient,
-		appKey.Namespace,
-		appKey.Name,
-	)
-	if err != nil {
-		return nil, fmt.Errorf(
-			"error finding Argo CD Application %q in namespace %q: %w",
-			appKey.Name, appKey.Namespace, err,
-		)
+	update *builtin.ArgoCDAppUpdate,
+) ([]*argocd.Application, error) {
+	namespace := update.Namespace
+	if namespace == "" {
+		namespace = libargocd.Namespace()
 	}
-	if app == nil {
-		return nil, fmt.Errorf(
-			"unable to find Argo CD Application %q in namespace %q",
-			appKey.Name, appKey.Namespace,
-		)
+
+	var apps []*argocd.Application
```

Comment:
Nit: Any reason this needs to be a slice of pointers? We could avoid iterating over the returned items from the list if we don't (and not have the possibility of `nil` pointers)